src/htm/algorithms/Connections.hpp

```diff
@@ -630,8 +631,8 @@ class Connections : public Serializable
   Permanence               connectedThreshold_; //TODO make const
 
   // Extra bookkeeping for faster computing of segment activity.
-  std::map<CellIdx, std::vector<Synapse>> potentialSynapsesForPresynapticCell_; //TODO use unordered_map
-  std::map<CellIdx, std::vector<Synapse>> connectedSynapsesForPresynapticCell_;
+  std::unordered_map<CellIdx, std::vector<Synapse>> potentialSynapsesForPresynapticCell_;
+  std::unordered_map<CellIdx, std::vector<Synapse>> connectedSynapsesForPresynapticCell_;
   std::map<CellIdx, std::vector<Segment>> potentialSegmentsForPresynapticCell_;
```

Comment:
Why not use `unordered_map` for all of the maps?

Reply:
> Why not use unordered_map for all of the maps?

I've tested and the performance was (small but mesurably) better for this case. 

unordered_map (HashMap): 
- avg O(1), but worst case O(n)
- mem >> map (tree) 

map (tree):
- guaranteed O(n*logn)
- lesser memory